sdk/search/Azure.Search.Documents/samples/Sample06_EncryptedIndex.md

```diff
@@ -0,0 +1,126 @@
+# Azure.Search.Documents Samples - Customer-managed Encryption Keys
+
+Azure Cognitive Search automatically encrypts indexed content at rest with [service-managed keys](https://docs.microsoft.com/azure/security/fundamentals/encryption-atrest#azure-encryption-at-rest-components). If more protection is needed, you can supplement default encryption with an additional encryption layer using keys that you create and manage in Azure Key Vault. This sample shows you how to set up and use customer-managed keys to further encrypt your data source connections, skillsets, indexers, and indexes for additional security.
+
+## Getting started
+
+To begin using Customer-Managed Keys (CMK), follow steps 1 through 5 to [configure customer-managed keys for data encryption](https://docs.microsoft.com/azure/search/search-security-manage-encryption-keys). This will guide you how to configure your Key Vault, register a service principal if necessary, and grant appropriate access to a service principal for a Key Vault key. This can be a key you created previously, or you can create a key that can wrap and unwrap service keys, including AES and RSA keys like in the following example:
+
+```bash
+# Create a resource group for your Key Vault and Cognitive Search service.
+# At this time, CMK is limited to certain regions including West US 2,
+# so we'll create everything in that location.
+az group create -n "<group-name>" -l westus2
+
+az keyvault create -g "<group-name>" -n "<vault-name>" -l westus2 --enable-purge-protection
+
+# Create a service principal for RBAC, or you can use Managed Identity.
+az ad sp create-for-rbac -n "<app-name>" --skip-assignment
+```
+
+The last command will provide output like the following:
+
+```json
+{
+    "appId": "<random-app-ID>",
+    "displayName": "<app-name>",
+    "name": "http://<app-name>",
+    "password": "<random-password>",
+    "tenant": "<tenant-ID>"
+}
+```
+
+Using that information, set up the Key Vault policy to allow specific access permissions and create a key:
+
+```bash
+az keyvault set-policy -n "<vault-name>" --spn "<generated-app-ID>" --certificate-permissions get --key-permissions get wrapKey unwrapKey --secret-permissions get
+
+# Create an RSA key that can wrap and unwrap service-managed keys.
+az keyvault key create --vault-name "<vault-name>" -n "<key-name>" --kty rsa --size 2048 --expires "<Y-m-d'T'H:M:S'Z'>" --ops wrapKey unwrapKey
+```
+
+## Using the encryption key
+
+Once you have a Key Vault key you can use for wrapping and unwrapping service keys, you can create a `SearchResourceEncryptionKey` with the information about the key and credentials for your service principal if not using Managed Identity.
+
+```C# Snippet:Azure_Search_Tests_Sample06_EncryptedIndex_CreateDoubleEncryptedIndex_Index
+// Create a credential to connect to Key Vault and use a specific key version created previously.
+SearchResourceEncryptionKey encryptionKey = new SearchResourceEncryptionKey(
```

Comment:
Why `SearchResourceEncryptionKey` instead of something more like `SearchIndexEncryptionKey`?  I'm guessing there's a conceptually important distinction I'm unaware of?  Is it maybe also encrypting projected data from skillsets?

Reply:
The same mechanism is used to protect synonym maps, and now other resource types as well. No point in replicating an identical type for each top-level resource of the data plane API.

Reply:
We GA'd with that type supporting indexes and synonym maps, so "resources" was decided as a more general term. We merely reused the type we already had when regenerating source to add this to more resource types.